File: project/choisujin-onboarding/src/main/java/com/innercircle/onboarding/common/response/ApiResponse.java
package com.innercircle.onboarding.common.response;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.innercircle.onboarding.common.exceptions.CommonException;
import lombok.Builder;
import lombok.Getter;

public class ApiResponse {

    @Getter
    @JsonInclude(JsonInclude.Include.NON_NULL)
    public static class Base<T> {

        private int status;
        private String code;
        private String message;
        private T data;

        public Base(T data) {
            this.status = ResponseStatus.SUCCESS.getHttpsStatus();
            this.code = ResponseStatus.SUCCESS.getCode();
            this.message = ResponseStatus.SUCCESS.getMessage();
            this.data = data;
        }

        public static <T> Base<T> success() {
            return new Base<>(null);
        }

        public static <T> Base<T> success(T data) {
            return new Base<>(data);
        }

    }

    @Getter
    @JsonInclude(JsonInclude.Include.NON_NULL)
    public static class Exception {

        private int status;
        private String code;
        private String message;
        private String path;
        private String description;

        @Builder
        public Exception(String path, CommonException commonException) {
            this.status = commonException.getResponse().getHttpsStatus();
            this.code = commonException.getResponse().getCode();
            this.message = commonException.getResponse().getMessage();
            this.path = path;
            this.description = commonException.getDescription();
        }

    }

}

File: project/kyungrok-onboarding/src/main/kotlin/kr/innercircle/onboarding/survey/dto/response/ApiResponse.kt
package kr.innercircle.onboarding.survey.dto.response

import com.fasterxml.jackson.annotation.JsonInclude
import kr.innercircle.onboarding.survey.exception.base.ErrorCode

/**
  * packageName : kr.innercircle.onboarding.survey.dto
  * fileName    : ApiResponse
  * author      : ckr
  * date        : 25. 6. 12.
  * description :
  */

@JsonInclude(JsonInclude.Include.NON_NULL)
data class ApiResponse(
    val success: Boolean = true,
    var data: Any? = null,
    var message: String? = null,
    var error: ApiErrorResponse? = null
) {
    constructor(errorCode: ErrorCode) : this(
        success = false,
        error = ApiErrorResponse(errorCode),
    )

    constructor(apiErrorResponse: ApiErrorResponse) : this(
        success = false,
        error = apiErrorResponse,
    )
}

data class ApiErrorResponse(
    val code: String,
    val message: String,
) {
    constructor(errorCode: ErrorCode) : this(
        code = errorCode.code,
        message = errorCode.message,
    )
}